controllers/vspherevm_controller.go

```diff
@@ -96,23 +96,13 @@ func AddVMControllerToManager(ctx context.Context, controllerManagerCtx *capvcon
 				&handler.EnqueueRequestForObject{},
 			),
 		).
-		WithEventFilter(predicates.ResourceNotPausedAndHasFilterLabel(mgr.GetScheme(), predicateLog, controllerManagerCtx.WatchFilterValue)).
+		WithEventFilter(predicates.ResourceHasFilterLabel(mgr.GetScheme(), predicateLog, controllerManagerCtx.WatchFilterValue)).
 		Watches(
 			&clusterv1.Cluster{},
 			handler.EnqueueRequestsFromMapFunc(r.clusterToVSphereVMs),
 			ctrlbldr.WithPredicates(
-				predicate.Funcs{
-					UpdateFunc: func(e event.UpdateEvent) bool {
-						newCluster := e.ObjectNew.(*clusterv1.Cluster)
-						// check whether cluster has either spec.paused or pasued annotation
-						return !annotations.IsPaused(newCluster, newCluster)
-					},
-					CreateFunc: func(e event.CreateEvent) bool {
-						cluster := e.Object.(*clusterv1.Cluster)
-						// check whether cluster has either spec.paused or pasued annotation
-						return annotations.IsPaused(cluster, cluster)
-					},
-				}),
+				predicates.ClusterPausedTransitionsOrInfrastructureReady(mgr.GetScheme(), predicateLog),
```

Comment:
Looks like the old predicate didn't make much sense?

We reconciled on:
* Update if the new cluster is not paused
* Create if the cluster is paused

I don't understand the latter

I would drop the prediate here. The old Update predicate basically let everything through, so I think we should continue that to avoid bad surprises